lib/SPIRV/libSPIRV/SPIRVModule.cpp

```diff
@@ -552,7 +553,8 @@ SPIRVModuleImpl::addEntry(SPIRVEntry *Entry) {
     } else
       IdEntryMap[Id] = Entry;
   } else {
-    EntryNoId.push_back(Entry);
+    if (EntryNoId.empty() || EntryNoId.back() != Entry)

```

Comment:
Why this condition? Probably we can use std::set to avoid duplicates.

Reply:
For example, entry of OpGroupDecorate will be added twice when decode SPIRV binary:
1) SPIRVGroupDecorateGeneric::decode(std::istream &I) -> Module->addGroupDecorateGeneric(this)
2) Decoder.getEntry() -> M.add(Entry)

Reply:
As far as  I understand with this condition the entry will be added to the vector if 1) the vector is empty or 2) if the last element of the vector equals to the entry. What if the entry is already in the vector but it is not at the end of the vector?